#116 Seller information tile

matgallacher opened the pull request:
Fixes #116

![image](https://github.com/user-attachments/assets/fa323cbb-2c79-4a99-8354-e96da617a281)

Product Details: Simple seller information tile

matgallacher commented on lib/features/products/seller_information.dart:
```diff
@@ -0,0 +1,95 @@
+        children: [
+          Expanded(
+            child: Row(
+              spacing: 8,
+              children: [
+                CircleAvatar(
+                  radius: 16,
+                  backgroundImage: AssetImage(AppImages.icProfile),
+                  foregroundImage: profileImage,
+                ),
```
Spacing is a valid property

CherryCIC commented:
Hey @matgallacher thank you for taking a look into this! 

As we discussed, the typography looks a lot nicer in Instrument Sans and we aim to stick with this font, but beyond that, I've got a review here. 
It still insisted that the spacing is invalid, but I took that out since we believe that's wrong.
Let me know if any of this is helpful, once this review is reviewed and the design has been updated, we can look to merge! Thanks again.

![image](https://github.com/user-attachments/assets/097788ed-a5e2-40b5-8b28-30109ee25be3)

---

### ✅ Visual & Functional Review (matches the screenshot)

* ✅ **Avatar, name, stars, charity logo, and “Ask seller” button** appear aligned and styled consistently with the UI shown.
* ✅ Ratings logic (full, half, empty stars) is correct.
* ✅ Button is clickable via `onAskSeller`, and profile image handling is well done.

---

### 🔍 Code Review Summary

#### 🧹 1. **Formatting & Syntax Issues**

* ❌ `EdgeInsetsGeometry.symmetric(...)` is **invalid**. Should be:

  ```dart
  padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 8),
  ```

* ❌ Multiple `Column(...)` and `Row(...)` blocks are duplicated at the end of the file — likely leftover copy-paste. **Remove all duplicates** from the end of the file starting from:

  ```dart
  Column(
    crossAxisAlignment: CrossAxisAlignment.start,
    ...


#### 📦 3. **Optional Suggestions**

* ✅ Use a helper method to abstract rating logic if reused often:

  ```dart
  List<Widget> _buildStars(double rating) {
    ...
  }
  ```

* ♻ Consider moving `OutlinedButton` to a reusable component if it's shared elsewhere.

---

### ✅ Final Verdict: Merge with Minor Fixes

> 🔧 **Fix immediately:**

* Replace `EdgeInsetsGeometry.symmetric` with `EdgeInsets.symmetric`.
* Remove duplicate `Column(...)` and `Row(...)` at the bottom.

Once the above is cleaned up, this can be **safely merged**.

matgallacher commented:
> Hey @matgallacher thank you for taking a look into this!
> 
> As we discussed, the typography looks a lot nicer in Instrument Sans and we aim to stick with this font, but beyond that, I've got a review here. It still insisted that the spacing is invalid, but I took that out since we believe that's wrong. Let me know if any of this is helpful, once this review is reviewed and the design has been updated, we can look to merge! Thanks again.

The default font on the Theme is set to Raleway, I think the Theme needs updating to use Instrument Sans, your theme should support your app not mean you have to fight it.

> ### 🔍 Code Review Summary
> #### 🧹 1. **Formatting & Syntax Issues**
> * ❌ `EdgeInsetsGeometry.symmetric(...)` is **invalid**. Should be:
>   ```dart
>   padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 8),

Fixed!

> * ❌ Multiple `Column(...)` and `Row(...)` blocks are duplicated at the end of the file — likely leftover copy-paste. **Remove all duplicates** from the end of the file starting from:
>   ```dart
>   Column(
>     crossAxisAlignment: CrossAxisAlignment.start,
>     ...
>   ```

Fixed!

yusuphjoluwasen commented on lib/features/products/seller_information.dart:
```diff
@@ -0,0 +1,95 @@
+                        WidgetStatePropertyAll(AppColors.greyTextColor),
+                  ),
+                  onPressed: onAskSeller,
+                  child: Text('Ask seller')),
```
Could you please move the 'Ask seller' text to AppStrings?

yusuphjoluwasen commented:
> > Hey @matgallacher thank you for taking a look into this!
> > As we discussed, the typography looks a lot nicer in Instrument Sans and we aim to stick with this font, but beyond that, I've got a review here. It still insisted that the spacing is invalid, but I took that out since we believe that's wrong. Let me know if any of this is helpful, once this review is reviewed and the design has been updated, we can look to merge! Thanks again.
> 
> The default font on the Theme is set to Raleway, I think the Theme needs updating to use Instrument Sans, your theme should support your app not mean you have to fight it.
> 
> > ### 🔍 Code Review Summary
> > #### 🧹 1. **Formatting & Syntax Issues**
> > 
> > * ❌ `EdgeInsetsGeometry.symmetric(...)` is **invalid**. Should be:
> >   ```dart
> >   padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 8),
> >   ```
> 
> Fixed!
> 
> > * ❌ Multiple `Column(...)` and `Row(...)` blocks are duplicated at the end of the file — likely leftover copy-paste. **Remove all duplicates** from the end of the file starting from:
> >   ```dart
> >   Column(
> >     crossAxisAlignment: CrossAxisAlignment.start,
> >     ...
> >   ```
> 
> I can't see this?!

The earlier font was Raleway, we’ll change the theme to use Instrument Sans in a different issue. You’re right, it looks better. Thanks
